### DISCUSSION TITLE
reading in a new ocean variable

### DISCUSSION THREAD

**Discussion by clittleaer** [NOAA-GFDL/MDTF-diagnostics#467] [Q&A] [answered]
Hi all, especially @jkrasting and @emaroon and @wrongkindofdoctor -- I'm getting started with reading in a new ocean variable (ZOS) using the example POD as a model. Only thing I want to do is read in a new model and obs dataset, and create a mean from a 5-year monthly timeslice. The input data is below:

(<xarray.Dataset>
 Dimensions:     (time: 300, nv: 2, xh: 1440, yh: 1080)
 Coordinates:
   * nv          (nv) float64 1.0 2.0
   * time        (time) object 1990-01-16 12:00:00 ... 2014-12-16 12:00:00
   * xh          (xh) float64 -299.7 -299.5 -299.2 -299.0 ... 59.53 59.78 60.03
   * yh          (yh) float64 -80.39 -80.31 -80.23 -80.15 ... 89.73 89.84 89.95
 Data variables:
     average_DT  (time) timedelta64[ns] dask.array<chunksize=(60,), meta=np.ndarray>
     average_T1  (time) datetime64[ns] dask.array<chunksize=(60,), meta=np.ndarray>
     average_T2  (time) datetime64[ns] dask.array<chunksize=(60,), meta=np.ndarray>
     time_bnds   (time, nv) object dask.array<chunksize=(60, 2), meta=np.ndarray>
     zos         (time, yh, xh) float32 dask.array<chunksize=(60, 1080, 1440), meta=np.ndarray>
 Attributes:
     filename:            ocean_monthly.199001-199412.zos.nc
     title:               CM4_historical
     associated_files:    areacello: 19900101.ocean_static.nc
     grid_type:           regular
     grid_tile:           N/A
     external_variables:  areacello,

I've renamed this file to zosgrtest.zos.mon.nc and created a dir structure following others. 
Here's the log: 

MDTF PACKAGE LOG

Started logging at 2023-05-26 07:10:23.604906
git hash/branch: f45de443b76947e3dbece10b0f553f3d4c743db5 (on mdsl)
sys.platform: 'linux'
sys.version: '3.10.11 | packaged by conda-forge | (main, May 10 2023, 18:58:44) [GCC 11.3.0]'
--------------------------------------------------------------------------------

07:10:23 DEBUG: log (logs.py line 328):
    Setting CASE_ROOT_DIR to MODEL_DATA_ROOT = '/glade/u/home/clittle/mdtf/inputdata/model'.

07:10:23 DEBUG: log (logs.py line 328):
    'sample_dataset' not supplied, using CASENAME = 'zosgrtest'.

07:10:23 DEBUG: log (logs.py line 328):
    Set program for <#9XGH:zosgrtest.zosexample> to 'python'.

07:10:23 WARNING: log (logs.py line 328):
    Deactivated <#9XIT:zosexample.zos> due to PodConfigEvent("Deactivating <#9XIT:zosexample.zos> due to variable name translation: "Standard name 'sea surface height above geoid' not defined in convention 'CMIP'.".").

07:10:23 INFO: log (logs.py line 328):
    Request for <#9XIT:zosexample.zos> (='(not translated)' @ 1mo) failed; looking for alternate data.

07:10:23 ERROR: log (logs.py line 328):
    Deactivated <#9XGH:zosgrtest.zosexample> due to PodDataError("Requested data not available for POD <#9XGH:zosgrtest.zosexample>: No alternate data available for <#9XIT:zosexample.zos>.").

07:10:23 ERROR: log (logs.py line 328):
    Deactivated <#9WF3:zosgrtest> due to ChildFailureEvent("Deactivating <#9WF3:zosgrtest> due to failure of all child objects.").

07:10:23 DEBUG: log (logs.py line 328):
    Writing file /glade/u/home/clittle/mdtf/wkdir/MDTF_zosgrtest_1990_1994/config_save.json

07:10:23 INFO: log (logs.py line 328):
    <#9WF3:zosgrtest>: Creating '/glade/u/home/clittle/mdtf/wkdir/MDTF_zosgrtest_1990_1994.tar'.

I am not sure whether this is because the code can't find and/or process the input data file, or because the variable name is problematic. Can anyone offer advice? Thanks, Chris

**Comment by jkrasting** [comment]
Hi @clittleaer - did you add `zos` to your `data/fieldlist_*.jsonc` entries?  If you push to a branch on GitHub we can take a closer look.

**Reply by clittleaer** [reply, marked as answer]
thanks @jkrasting the issue might have been in the "standard_name". I forgot the underscores :(

anyway on to the next battle
